test/test_page_without_a_file.rb

```diff
@@ -3,13 +3,13 @@
 require "helper"
 
 class TestPageWithoutAFile < JekyllUnitTest
-  def setup_page(*args)
+  def setup_page(*args, base: source_dir, klass: PageWithoutAFile)
```

Comment:
@ashmaroli Hey, sorry my first suggestion was unclear: I tought about moving `setup_page`, which is also defined in [TestPage](https://github.com/jekyll/jekyll/blob/73419cb374be1b8f45818a23116cf71db93549ce/test/test_page.rb#L6-L13), to [JekyllUnitTest](https://github.com/jekyll/jekyll/blob/master/test/helper.rb#L90) and give it then a `klass` option because that's the only difference between those two methods and then we would only define it once :)

Reply:
I see.. I should've understood your obvious intention.. my bad..

Reply:
No worries at all! It's already better 😉

Reply:
I don't want to change `test_page.rb` in this PR (since its an addition-only patch, right now..). Perhaps, a separate future PR would be best..

Reply:
Fair enough 👍